worker_types/gwci-linux/userdata

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
```

Comment:
this script is a good start but could use some love for reliability purposes.

to start, sprinkle a little of this:

```
#!/bin/bash
set -e
set -x

function retry {
  set +e
  local n=0
  local max=3
  while true; do
    "$@" && break || {
      if [[ $n -lt $max ]]; then
        ((n++))
        echo "Command failed. Attempt $n/$max:"
      else
        echo "Failed after $n attempts."
        exit 1
      fi
    }
  done
  set -e
}

exec &> /var/log/userdata.log
start_time="$(date '+%s')"
# do stuff
end_time="$(date '+%s')"
echo "UserData execution took: $(($end_time-$start_time)) seconds"
```

You can use `retry` like so: `retry apt install -y docker-ce`

Reply:
I like it!